### ISSUE TITLE
Incorrect links to index entries in language specification PDFs

### ISSUE THREAD

**Issue by eernstg** [dart-lang/site-www#1735]
On the page [Dart language specification](https://dart.dev/guides/language/spec) there are links to the [Dart 2.2 language specification](https://dart.dev/guides/language/specifications/DartLangSpec-v2.2.pdf) as well as a [bleeding edge version](https://spec.dart.dev/DartLangSpecDraft.pdf).

These are PDF documents, and at least the bleeding edge version is created using some automatic step (I'm not sure where this step is defined, but @mit-mit: I think it might be you who specified how to do this).

In any case, the index links in the resulting PDF documents do not work. For instance, going to page 206 (the last page) of the 2.2 specification and clicking 'static, 11' will take you to page 11, but the relevant piece of text is actually on page 15.

The `Makefile` that we're otherwise using in order to produce a PDF version of the specification will run the following commands:

```
pdflatex dartLangSpec.tex
makeindex dartLangSpec.idx
pdflatex dartLangSpec.tex
pdflatex dartLangSpec.tex
```

If the procedure used to create the PDFs on the 'Dart language specification' page omits some of the repetitions of the `pdflatex` command then the problem could be that the computation of labels and page numbers done by `pdflatex` (and stored in intermediate files named `*.ind` and `*.idx`, and maybe more) is not yet stable. In that case it should be sufficient to change the procedure such that it runs exactly those four commands shown above.

**Comment by kwalrath** [comment]
Unfortunately, @mit-mit is out and I don’t know how these docs are created.

**Comment by mit-mit** [comment]
I'm back! @eernstg here is the job that creates the bleeding edge spec:
https://github.com/dart-lang/language/blob/master/.travis.yml

It just calls make:
```
  - cd specification
  - make
  - mkdir firebase
  - cp dartLangSpec.pdf firebase/DartLangSpecDraft.pdf
```

Any idea why that might now work?

As for the stable spec, if you can generate a v2.4 spec, and updating the site is a simple PR to add that file [here](https://github.com/dart-lang/site-www/tree/master/src/_guides/language/specifications) and change the link to it [here](https://github.com/dart-lang/site-www/blob/master/src/_guides/language/spec.md).

**Comment by eernstg** [comment]
https://github.com/dart-lang/site-www/commit/f6d8ebaa25ebafa38da65b9b5b6aafa151bff181 updated the existing 2.2 spec to have correct index references.

**Comment by kwalrath** [comment]
Are we all good, or should we update the link in https://dart.dev/guides/language/spec (to 2.3 or 2.4)?

**Comment by eernstg** [comment]
2.3 added ui-as-code features (such as `if` and `for` in collections), and they are not in the spec yet, so it would not be correct to announce any available version of the spec as 2.3 or 2.4.

**Comment by kwalrath** [comment]
> 2.3 added ui-as-code features (such as `if` and `for` in collections), and they are not in the spec yet, so it would not be correct to announce any available version of the spec as 2.3 or 2.4.

I'm going to close this then. We can open another bug to update the spec when it's ready.

**Comment by eernstg** [comment]
There's one more thing, actually: We have fixed the issue with the numbered spec (2.2 at this point). But the issue still exists with the bleeding edge spec, and the procedure used to create that PDF will presumably continue to reproduce the issue each time a fresh bleeding edge spec is created (even though it's running exactly the same Makefile as the one that I'm using, and I don't experience the problem). So I think we should re-open this issue until we can fix that, too.

I suspect that the issue could be a discrepancy in the LaTeX/TeX libraries: If the commands executed by Travis for site-www are using an outdated version of some library (which causes the creation of these wrong index references) then that could explain why the same commands give different results on different computers. I do not expect `pdflatex` itself to behave differently, but it's probably much more likely that some imported libraries differ.

**Comment by eernstg** [comment]
Working as of https://github.com/dart-lang/language/commit/dc35fd957e7896946a54b4a14083ba9ca4a74f51.
